ring/lifecycler.go

```diff
@@ -759,11 +779,42 @@ func (i *Lifecycler) compareTokens(fromRing Tokens) bool {
 	return true
 }
 
+func (i *Lifecycler) waitBeforeJoining(ctx context.Context) error {
```

Comment:
I suggest to add a log (info level) at the beginning of the waiting period and one at the end.